### PR TITLE
SWATCH-2003: Enable wiremock testing in EE for azure

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: SWATCH-2003

## Description

Add flag to disable oidc auth on azure marketplace client for simplified EE testing

### CODE REVIEW

File: swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java [hunk 1]
```diff
@@ -58,13 +58,21 @@ public List<AzureClient> createClientForEachTenant() {
...
+                } else {
+                  api =
+                      QuarkusRestClientBuilder.newBuilder()
+                          .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()))
+                          .register(
+                              new AzureMarketplaceHeaderProvider(
+                                  azureMarketplaceProperties, client, oidcClients))
+                          .build(AzureMarketplaceApi.class);
+                }
                 return new AzureClient(client.getClientId(), api);
```
Comment (Sgitario) [inline swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java:76]:
```suggestion
                var apiBuilder = QuarkusRestClientBuilder.newBuilder()
                          .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()));
                if (!azureMarketplaceProperties.isDisableAzureOidc()) {
                  apiBuilder.register(
                              new AzureMarketplaceHeaderProvider(
                                  azureMarketplaceProperties, client, oidcClients));
                }
                return new AzureClient(client.getClientId(), apiBuilder.build(AzureMarketplaceApi.class));
```

I think it's clearer to use the builder in a common place instead.

**Review by Sgitario — COMMENTED** [review]
Small suggestion only.

File: swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java (hunk 1 above)
Comment (kflahert) [reply swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java]: Makes sense, thanks! updated

**Review by Sgitario — APPROVED** [review]

**Pull request merged**